lib/mutant/isolation.rb

```diff
@@ -47,11 +47,18 @@ def self.call(&block)
             end
 
             writer.close
-            Marshal.load(reader.read)
+            result = Marshal.load(reader.read)
+            if result.is_a?(Mutant::Result::Test)
```

Comment:
Also we have still to account for the fact that the subproject might NOT return such an object, as it crashed while producing it. We are on ruby a low-guarantee language.